File: Computing/MTR/2022/ACSBR/GET_ORDER_Nikola_406_20.py
import math

def get_price(option):
    cake_list = ["A", "B", "C", "D","E","F","G","H"]
    price_list = [25, 22, 38, 35, 15, 40, 53, 20]
    position = cake_list.index(option)
    return price_list[position]

def get_input():
    choice = ''
    while True:
        choice = input("Enter the choice of cake: ")
        if choice not in "ABCDEFGH":
            continue
        else:
            break

    return choice

def get_order():
    total = 0
    subtotal = 0
    gst = 0

    while True:
        choice = get_input()
        subtotal += get_price(choice)
        gst = 0.07 * subtotal
        total = subtotal + gst
        more = input("More purchase? Y or N: ")

        if more.upper() == "N":
            break
        else:
            pass

    print("\nSubtotal     $ {}".format(subtotal))
    print("GST          $ {}".format((round(gst, 2))))
    print("Total        $ {}".format(round(total, 2)))
    print("\nThank you!")

get_order()

File: Computing/C_Practice/revision/ACS/GET_ORDER_Nikola_406_20.c
#include <stdio.h>

char *cake_list = "ABCDEFGH";

int get_price(char option) {
    int cake_length = 8;
    int position = 0;
    int price_list[8] = {
        25,22,38,35,15,40,53,20
    };
    
    char *p;
    for (p = cake_list; *p != '\0'; p++) {
        if (*p == option) {
            break;
        } 
        position++;
    }

    return price_list[position];

}

char get_input() {
    char choice;

    char *p;
    for (;;) {
        printf("Enter the choice of cake: ");
        scanf(" %c", &choice);
        for (p = cake_list; *p != '\0'; p++) {
            if (*p == choice) {
                printf("%c\n", choice);
                return choice;
            }
        }
        printf("Enter an uppercase letter between A to H only\n");

    }
}

void get_order() {
    char choice;
    char more_purchase;

    int subtotal = 0;
    float gst;
    float total;

    for (;;) {
        choice = get_input();

        total += get_price(choice);
        subtotal = total;

        printf("More purchase? Y or N: ");
        scanf(" %c", &more_purchase);

        if (more_purchase == 'N') {
            break;
        }
        
    }

    gst = 0.07 * subtotal;
    total += gst;

    printf("Subtotal       $ %d\n", subtotal);
    printf("GST            $ %.2f\n", gst);
    printf("Total          $ %.2f\n", total);
}

int main() {
    get_order();

    return 0;
}
